tracee-rules/Makefile

```diff
@@ -27,11 +31,38 @@ $(OUT_RULES): $(GOSIGNATURES_DIR) $(REGO_SIGNATURES_SRC) | $(OUT_DIR)
 	mkdir -p $(OUT_RULES)
 	go build -buildmode=plugin -o $(OUT_GOSIGNATURES) $(GOSIGNATURES_SRC)
 	cp $(REGO_SIGNATURES_SRC) $(OUT_RULES)
-	
+
+check_%:
+	@command -v $* >/dev/null || (echo "missing required tool $*" ; false)
+
+tools = $(CMD_OPA) $(CMD_GO)
+.PHONY: $(tools)
+$(tools): % : check_%
+
 .PHONY: test
-test: $(GO_SRC)
+ifndef DOCKER
+test: $(GO_SRC) $(tools)
 	go test -v ./...
+	opa test . --verbose --ignore="examples" --ignore="dist"
+else
+test: $(DOCKER_BUILDER)
+	$(call docker_builder_make,$@)
+endif
 
 .PHONY: clean mostlyclean
 clean mostlyclean:
 	-rm -rf $(OUT_DIR)
+
+# docker_builder_make runs a make command in the tracee-builder container
+define docker_builder_make
+	$(CMD_DOCKER) run --rm \
+	-v $(abspath .):/tracee/tracee-rules \
+	-w /tracee/tracee-rules \
+	--entrypoint make $(DOCKER_BUILDER) $(1)
+endef
+
+# record built image id to prevent unnecessary building and for cleanup
```

Comment:
did you mean to also clean it up?

Reply:
not sure this was actually addressed

Reply:
As discussed: Need to add these in to the clean target in this makefile https://github.com/aquasecurity/tracee/blob/main/tracee-ebpf/Makefile#L149-L150